packages/rest/test/acceptance/file-upload/file-upload.acceptance.ts

```diff
@@ -0,0 +1,159 @@
+      .field('user', 'john')
+      .field('email', 'john@example.com')
+      .attach('certFile', path.resolve(FIXTURES, 'cert.pem'), {
+        filename: 'cert.pem',
+        contentType: 'multipart/form-data',
+      })
+      .expect(200);
+    expect(res.body.files[0]).containEql({
+      fieldname: 'certFile',
+      originalname: 'cert.pem',
+      mimetype: 'multipart/form-data',
+    });
+  });
+
+  function givenBodyParamController() {
+    const spec = anOpenApiSpec()
+      .withOperation('post', '/show-body', {
+        'x-operation-name': 'showBody',
+        requestBody: <RequestBodyObject>{
+          description: 'multipart/form-data value.',
+          required: true,
+          content: {
+            'multipart/form-data': {
+              // Skip body parsing
+              'x-parser': 'stream',
+              schema: {type: 'object'},
+            },
+          },
+        },
+        responses: {
+          200: {
+            content: {
+              'application/json': {
+                schema: {
+                  type: 'object',
+                },
+              },
+            },
+            description: '',
+          },
+        },
+      })
+      .build();
+
+    class RouteParamController {
+      async showBody(
+        request: Request,
+        @inject(RestBindings.Http.RESPONSE) response: Response,
+      ): Promise<Object> {
+        const storage = multer.memoryStorage();
+        const upload = multer({storage});
+        return new Promise<object>((resolve, reject) => {
+          upload.any()(request, response, err => {
+            if (err) reject(err);
+            else {
+              resolve({
+                files: request.files,
+                // tslint:disable-next-line:no-any
+                fields: (request as any).fields,
+              });
+            }
+          });
+        });
+      }
+    }
+
+    givenControllerClass(RouteParamController, spec);
+  }
+
+  function givenControllerClass(
+    // tslint:disable-next-line:no-any
+    ctor: new (...args: any[]) => Object,
+    spec: ControllerSpec,
+  ) {
+    handler.registerController(spec, ctor);
+  }
+
+  let rootContext: Context;
+  let handler: HttpHandler;
+  function givenHandler() {
+    rootContext = new Context();
+    rootContext.bind(SequenceActions.FIND_ROUTE).toProvider(FindRouteProvider);
+    rootContext
+      .bind(SequenceActions.PARSE_PARAMS)
+      .toProvider(ParseParamsProvider);
+    rootContext
+      .bind(SequenceActions.INVOKE_METHOD)
+      .toProvider(InvokeMethodProvider);
+    rootContext
+      .bind(SequenceActions.LOG_ERROR)
+      .to(createUnexpectedHttpErrorLogger());
+    rootContext.bind(SequenceActions.SEND).to(writeResultToResponse);
+    rootContext.bind(SequenceActions.REJECT).toProvider(RejectProvider);
+
+    rootContext.bind(RestBindings.SEQUENCE).toClass(DefaultSequence);
+    rootContext
+      .bind(RestBindings.REQUEST_BODY_PARSER)
+      .toClass(RequestBodyParser);
```

Comment:
Wow, that's a lot of work to setup the context. Since this is an acceptance-level test, can we use `RestApplication` please?

That way we can use regular `@requestBody` decorator and `app.controller` instead of `givenControllerClass`